### PR TITLE
fix(#391) - Included the error handling with the server and added retry options

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 
ONEPLAT-391

### 

# Explain the fix
Included the error handling with the server and added retry options with the gateway. When the request fails on the initial request and the error handling for unhandled requests.

## Does this PR introduce a breaking change
No

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Comment by sayak-sarkar** [comment]
@riginoommen Which issue is this resolving? #391 is a PR. Also, please follow the PR naming guidelines.

File: packages/api-gateway-service/package.json [hunk 1]
```diff
@@ -37,9 +37,10 @@
   "dependencies": {
     "@apollo/gateway": "0.16.6",
     "@types/jsonwebtoken": "8.5.0",
-    "@types/ws": "7.2.4",
-    "apollo-link-http": "1.5.17",
-    "apollo-link-ws": "1.0.20",
+    "@types/ws": "^7.2.4",
+    "apollo-link-http": "^1.5.17",
+    "apollo-link-retry": "^2.2.16",
+    "apollo-link-ws": "^1.0.20",
```
Comment (sayak-sarkar) [inline packages/api-gateway-service/package.json:43]:
```suggestion
    "@types/ws": "7.2.4",
    "apollo-link-http": "1.5.17",
    "apollo-link-retry": "2.2.16",
    "apollo-link-ws": "1.0.20",
```
Pin these dependencies!

**Comment by riginoommen** [comment]
> @riginoommen Which issue is this resolving? #391 is a PR. Also, please follow the PR naming guidelines.

Its the jira 391

File: packages/api-gateway-service/package.json (hunk 1 above)
Comment (riginoommen) [reply packages/api-gateway-service/package.json]: Updated

**Pull request merged**